app/src/main/java/com/hedvig/app/feature/insurance/ui/detail/coverage/PerilBottomSheet.kt

```diff
@@ -1,69 +1,190 @@
 package com.hedvig.app.feature.insurance.ui.detail.coverage
 
 import android.content.Context
+import android.graphics.drawable.PictureDrawable
 import android.os.Bundle
+import android.view.Gravity
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import android.widget.FrameLayout
+import android.widget.ImageView
 import androidx.core.os.bundleOf
+import com.bumptech.glide.RequestBuilder
+import com.google.android.material.bottomsheet.BottomSheetBehavior
+import com.google.android.material.bottomsheet.BottomSheetBehavior.BottomSheetCallback
+import com.google.android.material.bottomsheet.BottomSheetBehavior.STATE_DRAGGING
+import com.google.android.material.bottomsheet.BottomSheetDialog
 import com.google.android.material.bottomsheet.BottomSheetDialogFragment
 import com.hedvig.android.owldroid.fragment.PerilFragment
 import com.hedvig.app.BuildConfig
 import com.hedvig.app.R
 import com.hedvig.app.databinding.PerilBottomSheetBinding
+import com.hedvig.app.util.extensions.colorAttr
+import com.hedvig.app.util.extensions.dp
 import com.hedvig.app.util.extensions.isDarkThemeActive
+import com.hedvig.app.util.extensions.view.setHapticClickListener
 import com.hedvig.app.util.extensions.viewBinding
-import com.hedvig.app.util.svg.buildRequestBuilder
+import com.hedvig.app.util.safeLet
+import d
 import e
+import org.koin.android.ext.android.inject
+import java.util.ArrayList
 
 class PerilBottomSheet : BottomSheetDialogFragment() {
     private val binding by viewBinding(PerilBottomSheetBinding::bind)
+    private val requestBuilder: RequestBuilder<PictureDrawable> by inject()
     override fun onCreateView(
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
-    ): View =
-        inflater.inflate(R.layout.peril_bottom_sheet, container, false)
+    ): View = inflater.inflate(R.layout.peril_bottom_sheet, container, false)
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        val defaultStatusBarColor = dialog?.window?.statusBarColor
+        val defaultSystemUiVisibility = dialog?.window?.decorView?.systemUiVisibility
+        val bottomSheetDialog = super.onCreateDialog(savedInstanceState) as BottomSheetDialog
+        val title = requireArguments().getString(TITLE)
```

Comment:
Given the size of this argument list and the fact that they are all required, this might be a good use-case for `@Parcelize` and a `data class`.